cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -83,30 +88,35 @@ void sc_cleanup_invocation(sc_invocation *inv) {
         sc_cleanup_string(&inv->orig_base_snap_name);
         sc_cleanup_string(&inv->security_tag);
         sc_cleanup_string(&inv->executable);
+        sc_cleanup_string(&inv->rootfs_dir);
     }
 }
 
-void sc_maybe_pick_alt_base_snap(sc_invocation *inv) {
-    /* As a special fallback, allow the base snap to degrade from "core" to
-     * "ubuntu-core". This is needed for the migration tests. */
-    char mount_point[PATH_MAX] = {0};
-    sc_must_snprintf(mount_point, sizeof mount_point, "%s/%s/current/", SNAP_MOUNT_DIR, inv->base_snap_name);
-
-    if (access(mount_point, F_OK) == 0) {
+void sc_check_rootfs_dir(sc_invocation *inv) {
+    if (access(inv->rootfs_dir, F_OK) == 0) {
         return;
     }
-    /* The mount point is missing but for some snaps we may have an alternative. */
+
+    /* As a special fallback, allow the base snap to degrade from "core" to
+     * "ubuntu-core". This is needed for the migration from old
+     * ubuntu-core based systems to the new core.
+     */
     if (sc_streq(inv->base_snap_name, "core")) {
+        char mount_point[PATH_MAX] = {0};
+
         /* For "core" we can still use the ubuntu-core snap. This is helpful in
          * the migration path when new snap-confine runs before snapd has
          * finished obtaining the core snap. */
-        sc_must_snprintf(mount_point, sizeof mount_point, "%s/%s/current/", SNAP_MOUNT_DIR, "ubuntu-core");
+        sc_must_snprintf(mount_point, sizeof mount_point, "%s/%s/current", SNAP_MOUNT_DIR, "ubuntu-core");
```

Comment:
Do we have a check somewhere that will guarantee that SNAP_MOUNT_DIR is an absolute path (maybe in configure or somewhere). I suspect things would break in other ways, but better safe than sorry...

Reply:
I think we don't. We can add a trello card for this if you feel strongly about it.

Reply:
Like I said, things will likely break if it isn't, so I guess one could wait until that happens....